internal/types/search_filter.go

```diff
@@ -0,0 +1,156 @@
+package types
+
+import (
+	"time"
+
+	"github.com/samber/lo"
+
+	ierr "github.com/flexprice/flexprice/internal/errors"
+)
+
+// filtering options
+type DataType string
+
+const (
+	DataTypeString DataType = "STRING"
+	DataTypeNumber DataType = "NUMBER"
+	DataTypeDate   DataType = "DATE"
+	DataTypeArray  DataType = "ARRAY"
+)
+
+type FilterOperatorType string
+
+const (
```

Comment:
Switch to short lowercase slugs: eq, neq, contains, startsWith, endsWith, gt, lt, in, nin, before, after.
Keep the Go identifiers in PascalCase (Equals, Contains) and set the JSON tags to lowercase.